include/envoy/network/connection.h

```diff
@@ -120,14 +120,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
-
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
   /**
    * @return the local address of the connection. For client connections, this is the origin
    * address. For server connections, this is the local destination address. For server connections
    * it can be different from the proxy address if the downstream connection has been redirected or
    * the proxy is operating in transparent mode.
    */
   virtual const Address::Instance& localAddress() PURE;
+  virtual Address::InstanceConstSharedPtr localAddressConstSharedPtr() PURE;
```

Comment:
Going off-topic slightly, have we considered implementing all addresses as simply `__uint128_t`? That would greatly simplify `CidrRange` operations. @jamessynge